contentcuration/contentcuration/viewsets/contentnode.py

```diff
@@ -905,23 +905,43 @@ def copy(
             error = ValidationError("Copy pk already exists")
             return str(error)
 
-        task_args = {
-            "user_id": self.request.user.id,
-            "channel_id": channel_id,
-            "source_id": source.id,
-            "target_id": target.id,
-            "pk": pk,
-            "mods": mods,
-            "excluded_descendants": excluded_descendants,
-            "position": position,
-        }
-
-        task, task_info = create_async_task(
-            "duplicate-nodes", self.request.user, **task_args
+        can_edit_source_channel = ContentNode.filter_edit_queryset(
+            ContentNode.objects.filter(id=source.id), user=self.request.user
+        ).exists()
+
+        def update_progress(meta=None):
+            if meta:
+                if "progress" in meta:
+                    Change.create_change(
+                        generate_update_event(
+                            pk,
+                            CONTENTNODE,
+                            {COPYING_PROGRESS: meta["progress"]},
+                            channel_id=channel_id
+                        ),
+                        created_by_id=self.request.user.id,
+                        applied=True
+                    )
```

Comment:
It looks like this approach will be okay for copying, but it doesn't look like it will work for publishing and may confuse devs in the future if we carry on using this pattern. The reason is that if the task requires a database transaction, then these writes for progress could be contained within the transaction and won't be readable outside of the transaction until it is committed, obviously. Looking at `map_content_nodes` that is called during publishing, it seems that it would run into this problem.

For this reason, in my task-related WIP, I utilized the task events to broadcast progress to a listener that could read and write outside of any database transactions occurring within the task.